activity.py

```diff
@@ -42,37 +41,34 @@ def build_toolbar(self):
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
 
-        separator = Gtk.SeparatorToolItem()
-        separator.props.draw = False
-        separator.set_expand(False)
-        toolbar_box.toolbar.insert(separator, -1)
-        separator.show()
+        self._add_toolbar_separator(toolbar_box)
 
-        button = ToolButton('speaker-muted-100')
-        button.set_tooltip(_('Sound'))
-        button.connect('clicked', self.sound_control)
-        toolbar_box.toolbar.insert(button, -1)
+        sound_button = ToolButton('speaker-muted-100')
+        sound_button.set_tooltip(_('Sound'))
+        sound_button.connect('clicked', self._sound_control)
+        toolbar_box.toolbar.insert(sound_button, -1)
 
-        separator = Gtk.SeparatorToolItem()
-        separator.props.draw = False
-        separator.set_expand(True)
-        toolbar_box.toolbar.insert(separator, -1)
-        separator.show()
+        self._add_toolbar_separator(toolbar_box, expand=True)
 
         stop_button = StopButton(self)
         toolbar_box.toolbar.insert(stop_button, -1)
         stop_button.show()
         stop_button.connect('clicked', self._stop_cb)
 
-    def sound_control(self, button):
-        self.sound = not self.sound
-        self.game.sound = self.sound
-        if not self.sound:
-            button.set_icon_name('speaker-muted-000')
-            button.set_tooltip(_('No sound'))
-        else:
-            button.set_icon_name('speaker-muted-100')
-            button.set_tooltip(_('Sound'))
+    def _add_toolbar_separator(self, toolbar_box, expand=False):
+        separator = Gtk.SeparatorToolItem()
+        separator.props.draw = False
+        separator.set_expand(expand)
+        toolbar_box.toolbar.insert(separator, -1)
+        separator.show()
+
+    def _sound_control(self, button):
+        self.sound_enabled = not self.sound_enabled
+        self.game.sound = self.sound_enabled
+        icon_name = 'speaker-muted-000' if not self.sound_enabled else 'speaker-muted-100'
+        tooltip = _('No sound') if not self.sound_enabled else _('Sound')
+        button.set_icon_name(icon_name)
+        button.set_tooltip(tooltip)
```

Comment:
This doesn't really make the code better in terms of readability, lots of things have been packed into one line so I don't agree with the change, although what could make it better is improving on the use of `self.sound` in this function as I don't see a reason why it's used the way it's used here.